Samples/HelloWorld/OrleansClient/Program.cs

```diff
@@ -16,8 +16,27 @@ public class Program
         static int Main(string[] args)
         {
             var config = ClientConfiguration.LocalhostSilo();
-            const int initializeAttemptsBeforeFailing = 5;
-            int attemp = 0;
+            try
+            {
+                InitializeWithRetries(config, initializeAttemptsBeforeFailing: 5);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Orleans client initialization failed failed due to {ex}");
+
+                Console.ReadLine();
+                return 1;
+            }
+
+            DoClientWork().Wait();
+            Console.WriteLine("Press Enter to terminate...");
+            Console.ReadLine();
+            return 0;
+        }
+
+        private static void InitializeWithRetries(ClientConfiguration config, int initializeAttemptsBeforeFailing)
+        {
+            int attempt = 0;
             while (true)
             {
```

Comment:
not sure I understand what's wrong there... an Exception of a type other than SiloUnavailableException will bubble up, but not be retried. We are catching it in Main so that we print an error and exit instead of letting the exception crash the app